//syntax = "proto3";
edition = "2023";

package storage;

// In-progress transactions are put & deleted with the "last write tx open" pointer as user-defined timestamp.
message InProgress {
    uint64 txId = 1;
}

// Committed transactions are written so that conficts can be detected with to-be-committed transactions.
message Transaction {
    uint64 txId = 1;
    repeated bytes touched = 2;
}

message TransactionManagerState {
    uint64 lastTxId = 1;
    repeated uint64 committed = 2;
    repeated uint64 invalid = 4;
}

